[BUGFIX] Remove vestigial path normalization in chains

mmun opened the pull request:
This fixes a bug that occurs when chaining through a property that starts with a capital letter.

[Here is a JSBin](http://emberjs.jsbin.com/veromo/1/edit?html,js,output) with a demonstration of the bug.

mixonic commented on packages/ember-metal/lib/binding.js:
```diff
@@ -169,19 +162,45 @@ Binding.prototype = {
+        deprecate(
+          `Connecting an Ember.Binding to a global path such as '${this._from}' is deprecated. Please use a service instead.`,
+          false,
+          { id: 'ember-metal.global-bindings', until: '3.0.0' }
```
Can you please make an entry in the [deprecation guide](https://github.com/emberjs/website/blob/master/source/deprecations/v2.x.html.md) and add a `url` property here.

mmun replied:
Let's discuss whether we want to deprecate all of Ember.Binding first.

mmun replied:
(On that note I am going to remove the deprecation from this PR)

mmun committed:
Reimplement Ember.Binding to avoid global chains

mmun committed:
[BUGFIX] Remove vestigial path normalization in chains

Removes support for paths containing globals and "this". These were
intended to be removed at the time when globals support was removeded
from Ember.get/set and from inside of temlates, e.g.

```hbs
{{SomeGlobal.foo}}
```

mmun committed:
Remove "pending flush" concept from chains

mmun commented:
Kris pointed out a couple other things that can be cleaned up:

- https://github.com/emberjs/ember.js/blob/59fa9721c2278d2d23d4d4417b00ff726e60167d/packages/ember-metal/lib/chains.js#L153
- https://github.com/emberjs/ember.js/blob/59fa9721c2278d2d23d4d4417b00ff726e60167d/packages/ember-metal/lib/chains.js#L335

And we should do some performance testing to see how much of a win it is.

stefanpenner commented:
:fire:

rwjblue commented:
Thanks @mmun!

stefanpenner commented on packages/ember-metal/lib/binding.js:
```diff
@@ -219,20 +232,20 @@ Binding.prototype = {
 
   /* Called when the from side changes. */
   fromDidChange(target) {
```
this arg doesn't seem one used anymore?

krisselden commented:
This should be documented as breaking but since the behavior made no sense and didn't align with templates I doubt this will bite anyone
